core/variant/typed_array.h

```diff
@@ -46,10 +46,15 @@ class TypedArray : public Array {
 		_ref(p_array);
 	}
 	_FORCE_INLINE_ TypedArray(const Variant &p_variant) :
-			Array(Array(p_variant), Variant::OBJECT, T::get_class_static(), Variant()) {
+			TypedArray(Array(p_variant)) {
 	}
-	_FORCE_INLINE_ TypedArray(const Array &p_array) :
-			Array(p_array, Variant::OBJECT, T::get_class_static(), Variant()) {
+	_FORCE_INLINE_ TypedArray(const Array &p_array) {
+		set_typed(Variant::OBJECT, T::get_class_static(), Variant());
+		if (is_same_typed(p_array)) {
+			_ref(p_array);
+		} else {
+			assign(p_array);
+		}
```

Comment:
Alternatively this block can be replaced by:
```cpp
		if (p_array.get_typed_builtin() == Variant::OBJECT && p_array.get_typed_class_name() == T::get_class_static() && p_array.get_typed_script() == Variant()) {
			_ref(p_array);
		} else {
			set_typed(Variant::OBJECT, T::get_class_static(), Variant());
			assign(p_array);
		}
```
To avoid doing the typing part, though in either case it will allocate private data possibly unnecessarily

Reply:
> though in either case it will allocate private data possibly unnecessarily

Then, I would keep the simplest version.

Reply:
Then I will and a future cleanup or improvement can hand that